tabby-ssh/src/components/sftpPanel.component.ts

```diff
@@ -106,17 +109,23 @@ export class SFTPPanelComponent {
         }
     }
 
+    async openCreateDirectoryModal (): Promise<void> {
+        const modal = this.ngbModal.open(SFTPCreateDirectoryModalComponent)
+        const directoryName = await modal.result
+        if (directoryName !== '') {
+            this.sftp.mkdir(path.join(this.path, directoryName)).finally(() => {
```

Comment:
should be `then` (or better, just `await`) to prevent nagivating into a folder that failed to get created